plugins/workspace-minimap/src/positioned_minimap.ts

```diff
@@ -14,23 +14,131 @@ import {Minimap} from './minimap';
 
 /**
  * A positionable version of minimap that implements IPositionable.
- * @param {Minimap} minimap The minimap to position.
- * @implements {Blockly.IPositionable}
  */
-export class PositionedMinimap {
+export class PositionedMinimap implements Blockly.IPositionable {
     protected minimap: Minimap;
+    protected primaryWorkspace: Blockly.WorkspaceSvg;
+    protected margin: number;
+    protected top: number;
+    protected left: number;
+    protected width: number;
+    protected height: number;
+    id: string;
     /**
-     * Constructor for a minimap
-     * @param {Minimap} minimap The workspace to sit in.
+     * Constructor for a positionable minimap.
+     * @param workspace The workspace to mirror.
      */
-    constructor(minimap: Minimap) {
-      this.minimap = minimap;
+    constructor(workspace: Blockly.WorkspaceSvg) {
+      this.primaryWorkspace = workspace;
+      this.minimap = new Minimap(this.primaryWorkspace);
+      this.id = 'minimap';
+      this.margin = 20;
+      this.top = 0;
+      this.left = 0;
+      this.width = 225; // TODO: Bumping size responsiveness to future branch.
+      this.height = 150;
     }
 
     /**
      * Initialize.
      */
     init(): void {
-    //   this.work
+      this.primaryWorkspace.getComponentManager().addComponent({
+        component: this,
+        weight: 3,
+        capabilities: [Blockly.ComponentManager.Capability.POSITIONABLE],
+      });
+      this.minimap.init();
+      this.setMinimapCss();
+      this.primaryWorkspace.resize();
+      Blockly.svgResize(this.minimap.minimapWorkspace);
+    }
+
+    /**
+     * Returns the bounding rectangle of the UI element in pixel units
+     * relative to the Blockly injection div.
+     * @returns The component’s bounding box.
+     */
+    getBoundingRectangle(): Blockly.utils.Rect {
+      return new Blockly.utils.Rect(
+          this.top, this.top + this.height,
+          this.left, this.left + this.width);
+    }
+
+    /**
+     * Positions the minimap.
+     * @param metrics The workspace metrics.
+     * @param savedPositions List of rectangles already on the workspace.
+     */
+    position(metrics: Blockly.MetricsManager.UiMetrics,
+        savedPositions: Blockly.utils.Rect[]): void {
+      const hasVerticalScrollbars = this.primaryWorkspace.scrollbar &&
+          this.primaryWorkspace.scrollbar.canScrollHorizontally();
+      const hasHorizontalScrollbars = this.primaryWorkspace.scrollbar &&
+          this.primaryWorkspace.scrollbar.canScrollVertically();
```

Comment:
Are these checking the correct scrollbars? I know this matches what the backpack is doing, but I am a bit confused haha.

Reply:
Yes, they are checking the scrollbars in the primary workspace. It needs that to offset the location of the minimap in case they are present.

Reply:
I'm just confused why `hasVerticalScrollbars` is based on `canScrollHorizontally`. I'm pretty sure the vertical scrollbars affect the vertical scroll.

Reply:
Ok, so this is in fact an error. I've corrected it so that vertical goes with vertical and vise versa. Also, I added another use case that was no previously checked. If the scrollbar exists and scrolling is enabled, but the scrollbar is not visible, it should adjust for that.

Reply:
Please also file a bug to update the backpack plugin similarly!